pkg/network/pod/annotations/generator.go

```diff
@@ -107,12 +105,7 @@ func (g Generator) GenerateFromActivePod(vmi *v1.VirtualMachineInstance, pod *k8
 		return iface.SRIOV != nil || vmispec.HasBindingPluginDeviceInfo(iface, g.clusterConfig.GetNetworkBindings())
 	})
 
-	networkStatusAnnotation := pod.Annotations[networkv1.NetworkStatusAnnot]
-	networkDeviceInfoMap, err := deviceinfo.MapNetworkNameToDeviceInfo(vmi.Spec.Networks, networkStatusAnnotation, ifaces)
-	if err != nil {
-		log.Log.Warningf("failed to create network device-info-map: %v", err)
-	}
-
+	networkDeviceInfoMap := deviceinfo.MapNetworkNameToDeviceInfo(vmi.Spec.Networks, ifaces, multus.NetworkStatusesFromPod(pod))
```

Comment:
Following this PR, only `multus.NetworkStatusesFromPod` does the actual parsing / unmashalling.
In case there is an umarshaling error, there will be a log entry and a nil slice will be returned.

Reply:
@nirdothan , please feedback if you got convinced or not so this thread can continue or reach resolution.